### ISSUE TITLE
tombo resquiggle  cpu use question

### ISSUE THREAD

**Issue by hygine** [nanoporetech/tombo#166]
when i run tombo resquiggle, I set the --processes  2 , but when  it's  run , the message  of cpu use  more than 5 , what's the problem？

![image](https://user-images.githubusercontent.com/11896993/55942460-00745b80-5c77-11e9-8698-bbc598afa3ae.png)
![image](https://user-images.githubusercontent.com/11896993/55942486-09fdc380-5c77-11e9-9a97-15ab2dbb187a.png)

**Comment by hygine** [comment]
@rmp

**Comment by marcus1487** [comment]
Tombo spawns a number of processes to handle different data management and IO tasks (sending read files to processes, reporting progress, collating read index information). These data management tasks should use very little computational resources. Then for each computational process requested, a new process and a new thread are spawned. The thread handles the mapping via the mappy interface as this requires the loaded genome to reside in a shared memory space. The process performs the re-squiggling logic as this needs large amounts of computation and must avoid the GIL.

In short, tombo should not use more resources than requested (some caveats in #101 ), but does spawn more processes than requested for data management. You can see the code that handles the spawning and joining of these tasks [here in the code](https://github.com/nanoporetech/tombo/blob/master/tombo/resquiggle.py#L1845) if you are interested further.

**Comment by marcus1487** [comment]
Ah looking closer at the `top` results sent this is actually probably the issue raise in #101 . I would recommend setting `export OMP_NUM_THREADS=1` in your environment to keep numpy from using more resources than requested.
